### PR TITLE
Add API docs for StaticArray

### PR DESCRIPTION
**Pull request by straight-shoota**

### CODE REVIEW

File: src/static_array.cr [hunk 1]
```diff
@@ -1,4 +1,26 @@
 # A fixed-size, stack allocated array.
+#
+# `StaticArray` is a generic type with type argument `T` specifying the type of
+# it's elements and `N` the fixed size. For example `StaticArray(Int32, 3)`
```
Comment (r00ster91) [inline src/static_array.cr]: I think "its" instead of "it's" is correct here.

File: src/static_array.cr [hunk 2]
```diff
@@ -1,4 +1,26 @@
 # A fixed-size, stack allocated array.
+#
+# `StaticArray` is a generic type with type argument `T` specifying the type of
+# its elements and `N` the fixed size. For example `StaticArray(Int32, 3)`
```
Comment (vlazar) [inline src/static_array.cr]: Shouldn't it be
> its elements and `N` **is** the fixed size.

**Comment by straight-shoota** [comment]
I added an example for use with `uninitialized`. @RX14 @jhass could you pleas re-review?

File: src/static_array.cr (hunk 2 above)
Comment (vlazar) [reply src/static_array.cr]: @straight-shoota Thanks for explanation. TIL.

**Pull request merged**